lib/iris/experimental/regrid.py

```diff
@@ -1312,3 +1313,401 @@ def regridder(self, src_grid, target_grid):
 
         """
         return _CurvilinearRegridder(src_grid, target_grid, self.weights)
+
+
+class _ProjectedUnstructuredRegridder(object):
+    """
+    This class provides a wrapper for the calculation of
+    scipy.interpolate.griddata.
+
+    """
+    def __init__(self, src_cube, tgt_grid_cube, method,
+                 projection=None):
+        """
+        Create a regridder for conversions between the source
+        and target grids.
+
+        Args:
+
+        * src_cube:
+            The :class:`~iris.cube.Cube` providing the source points.
+        * tgt_grid_cube:
+            The :class:`~iris.cube.Cube` providing the target grid.
+        * method:
+            Either 'linear' or 'nearest'.
+        * projection:
+            The projection in which the interpolation is performed, or None. If
+            None, a XXX projection is used.
+
+        """
+        # Validity checks.
+        if not isinstance(src_cube, iris.cube.Cube):
+            raise TypeError("'src_cube' must be a Cube")
+        if not isinstance(tgt_grid_cube, iris.cube.Cube):
+            raise TypeError("'tgt_grid_cube' must be a Cube")
+
+        # Snapshot the state of the target cube to ensure that the regridder
+        # is impervious to external changes to the original source cubes.
+        self._tgt_grid = snapshot_grid(tgt_grid_cube)
+
+        # Check the target grid units.
+        for coord in self._tgt_grid:
+            RectilinearRegridder._check_units(coord)
```

Comment:
I'm not sure I like the idea of calling a static method of an unrelated class here, especially since the method is private. It should probably be in a common base class, but maybe this is ok for now...